### DISCUSSION TITLE
Updating dcarte Library

### DISCUSSION THREAD

**Discussion by francescapalermo** [esoreq/dcarte#6] [Q&A]
I am updating the dcarte dataset on my machine and I am looping over the key,values in the domains iteritems
At the moment I am only updating RAW and BASE recipes with the following command:

`dcarte.load(value, key, update=True, delay=0)`

where `value` is for example `Behavioural` and `key` is `RAW`.
The main problem is that it always gets stuck on processing `encounter`
So I was wondering if it is indeed possible to update all the libraries or if some libraries should be skipped

```
Finished Loading Encounter in:                12.8 seconds   
Processing encounter :*
```

Moreover, is it possible to update also the PROFILE AND BED_HABITS recipes with the same dcarte.load command or should the methods in `profile.py` be run again?

**Comment by francescapalermo** [comment]
SOLUTION:
We discussed on Slack with Eyal. Below a summary of the conversation to help you update the dataset

EYAL is working on implementing a function called dcarte.update_all() which, when run, will update the whole database. 
FRA: In the meantime you can update the current dataset using the following code which will update the RAW and BASE databases. You can try removing the check for RAW and BASE to test it with the rest of domains (e.g. Weekly_Profile)

```
domains = dcarte.domains()
print('Update Database')
for key, values in domains.iteritems():
    if key == 'RAW' or key == 'BASE':
        #print(key, values)
        for value in values:
            if value =='' or value == 'Encounter':
                continue
            print(key, value)
            data = dcarte.load(value, key)
            if 'start_date' in data.columns:
                if data['start_date'].dt.floor('D').max() == today:
                    print('Already Updated')
                    del data
                    print('-----')
                    continue
            del data
            dcarte.load(value, key, update=True)
            print('-----')
```

**Reply by esoreq** [reply]
So I just released 0.3.32 it has two new functions 
`dcarte.update_raw()`
and `dcarte.update_domain(domain='base')`
The domain can be any domain and the function will go over **the dataset you downloaded only** and update them
